app/liquid_tags/stackexchange_tag.rb

```diff
@@ -42,59 +46,54 @@ def render(_context)
 
   private
 
-  def valid_site?(site)
-    (site =~ /[a-z.]+/i)&.zero?
-  end
-
   def parse_site(input)
-    return "stackoverflow" if tag_name == "stackoverflow"
-
-    site = input.match(/[a-z.]+/i)[0]
-    unless valid_site?(site)
-      raise StandardError, I18n.t("liquid_tags.stackexchange_tag.invalid_site", tag: tag_name, input: input)
-    end
-
-    site
-  end
+    return "stackoverflow" if tag_name == "stackoverflow" || input.match?(STACKOVERFLOW_REGEXP)
 
-  def valid_input?(input)
-    return false if input.nil?
+    match = pattern_match_for(input, REGEXP_OPTIONS)
+    # rubocop:disable Layout/LineLength
+    raise StandardError, I18n.t("liquid_tags.stackexchange_tag.invalid_site", tag: tag_name, input: input) unless match && match_group_present?(match, "subdomain")
+    # rubocop:enable Layout/LineLength
 
-    ID_REGEXP.match?(input.split.first)
+    match[:subdomain].downcase
   end
 
-  def handle_response_error(response, input)
-    raise StandardError, "Calling StackExchange API failed: #{response&.error_message}" if response.code != 200
-
-    return unless response["items"].length.zero?
-
-    raise StandardError, I18n.t("liquid_tags.stackexchange_tag.post_not_found", tag: tag_name, input: input)
+  def match_group_present?(match, group_name)
+    match.names.include?(group_name)
   end
 
   def get_data(input)
-    unless valid_input?(input)
-      raise StandardError, I18n.t("liquid_tags.stackexchange_tag.invalid_id", tag: tag_name, input: input)
-    end
-
-    id = input.split.first
+    id, = input.split
+    match = pattern_match_for(id, REGEXP_OPTIONS)
+    # rubocop:disable Layout/LineLength
+    raise StandardError, I18n.t("liquid_tags.stackexchange_tag.invalid_id", tag: tag_name, input: input) unless match && match_group_present?(match, "id")
+    # rubocop:enable Layout/LineLength
 
-    url = "#{API_URL}posts/#{id}?site=#{@site}&filter=#{FILTERS['post']}" \
+    url = "#{API_URL}posts/#{match[:id]}?site=#{@site}&filter=#{FILTERS['post']}" \
           "&key=#{ApplicationConfig['STACK_EXCHANGE_APP_KEY']}"
     post_response = HTTParty.get(url)
 
     handle_response_error(post_response, input)
 
     @post_type = post_response["items"][0]["post_type"]
 
-    url = "#{API_URL}#{@post_type.pluralize}/#{id}?site=#{@site}" \
+    url = "#{API_URL}#{@post_type.pluralize}/#{match[:id]}?site=#{@site}" \
           "&filter=#{FILTERS[@post_type]}&key=#{ApplicationConfig['STACK_EXCHANGE_APP_KEY']}"
     final_response = HTTParty.get(url)
 
     handle_response_error(final_response, input)
 
-    final_response["items"][0]
+    [final_response["items"][0], @post_type]
+  end
+
+  def handle_response_error(response, input)
+    raise StandardError, "Calling StackExchange API failed: #{response&.error_message}" if response.code != 200
```

Comment:
HTTParty defines predicate methods for the different HTTP status codes, so you can use `ok?` instead of checking for `200` explicitly. 

```suggestion
    raise StandardError, "Calling StackExchange API failed: #{response&.error_message}" unless response.ok?
```